File: Java Word Examples/ShapesApiDemo.java
import spire.cloud.word.sdk.client.*;
import spire.cloud.word.sdk.client.api.ShapesApi;
import spire.cloud.word.sdk.client.model.*;
import java.util.List;

public class ShapesApiDemo {
    static String appId = "your id";
    static String appKey = "your key";
    static String baseUrl = "https://api.e-iceblue.cn";
    static Configuration wordConfiguration = new Configuration(appId, appKey, baseUrl);
    static ShapesApi shapesApi = new ShapesApi(wordConfiguration);

    public static void addShape() throws ApiException {
        String name = "addShape.docx";
        String paragraphPath = "Section/0/Body/0/Paragraph/0";
        int indexInParagraph = 1;
        String folder = "input";
        String storage = null;
        String password = null;
        ShapeFormat shapeFormat = new ShapeFormat(40f, 40f, ShapeFormat.ShapeTypeEnum.RECTANGLE);
        shapeFormat.setHorizontalOrigin(ShapeFormat.HorizontalOriginEnum.PAGE);
        shapeFormat.setVerticalOrigin(ShapeFormat.VerticalOriginEnum.PARAGRAPH);
        shapeFormat.setVerticalPosition(50f);
        shapeFormat.setHorizontalPosition(150f);
        Color color = new Color(0, 206, 209);
        shapeFormat.setFillColor(color);
        shapeFormat.setRotation(45f);
        shapeFormat.setStrokeWeight(2f);
        Color color_2 = new Color(173, 255, 47);
        shapeFormat.setStrokeColor(color_2);
        shapeFormat.setTextWrappingType(ShapeFormat.TextWrappingTypeEnum.BOTH);
        shapeFormat.setTextWrappingStyle(ShapeFormat.TextWrappingStyleEnum.INFRONTOFTEXT);
        shapeFormat.setZOrder(1);
        ShapeFormat shapeProperties = shapeFormat;
        String destFilePath = "output/addShape_output.docx";
        shapesApi.addShape(name, paragraphPath, shapeFormat, destFilePath,folder, storage, indexInParagraph, password);
    }

    public static void deleteShape() throws ApiException {
        String name = "deleteShape.docx";
        String paragraphPath = "Section/0/Body/0/Paragraph/0";
        Integer index = 0;
        String folder = "input";
        String storage = null;
        String password = null;
        String destFilePath = "output/deleteShape_output.docx";
        shapesApi.deleteShape(name, paragraphPath, index, destFilePath, folder, storage, password);
    }
    public static void getShapeFormat() throws ApiException {
        String name = "template.docx";
        String paragraphPath = "Section/0/Body/0/Paragraph/0"; ;
        Integer index = 3;
        String folder = "input";
        String storage = null;
        String password = null;
        ShapeFormat response = shapesApi.getShapeFormat(name, paragraphPath, index, folder, storage, password);
    }
    public static void getShapes() throws ApiException {
        String name = "getShapes.docx";
        String folder = "input";
        String storage = null;
        String password = null;
        String paragraphPath = null;
        List<ObjectInfo> response = shapesApi.getShapes(name, folder, storage, password, paragraphPath);
    }
    public static void updateShapeFormat() throws ApiException {
        String name = "updateShapeFormat.docx";
        String paragraphPath = "Section/0/Body/0/Paragraph/0";
        Integer index = 0;
        String folder = "input";
        String storage = null;
        String password = null;
        ShapeFormat shapeFormat = new ShapeFormat(50f, 50f, ShapeFormat.ShapeTypeEnum.OCTAGON);
        shapeFormat.setHorizontalOrigin(ShapeFormat.HorizontalOriginEnum.LEFTMARGINAREA);
        shapeFormat.setVerticalOrigin(ShapeFormat.VerticalOriginEnum.PARAGRAPH);
        shapeFormat.setVerticalPosition(80f);
        shapeFormat.setHorizontalPosition(80f);
        Color color = new Color(100, 100, 100);
        shapeFormat.setFillColor(color);
        shapeFormat.setRotation(0f);
        shapeFormat.setStrokeWeight(2f);
        Color color_2 = new Color(0, 206, 209);
        shapeFormat.setStrokeColor(color_2);
        shapeFormat.setTextWrappingType(ShapeFormat.TextWrappingTypeEnum.BOTH);
        shapeFormat.setTextWrappingStyle(ShapeFormat.TextWrappingStyleEnum.INFRONTOFTEXT);
        shapeFormat.setZOrder(1);
        ShapeFormat shapeProperties = shapeFormat;
        String destFilePath = "output/updateShapeFormat_output.docx";
        shapesApi.updateShapeFormat(name, paragraphPath, index, shapeFormat, destFilePath, folder, storage, password);
    }
}

File: C# Word Examples/ShapesApiDemo.cs
using Spire.Cloud.Word.Sdk.Api;
using Spire.Cloud.Word.Sdk.Client;
using Spire.Cloud.Word.Sdk.Model;
using System;

namespace CloudWord
{
    class ShapesApiDemo
    {
        static string appId = "your id";
        static string appKey = "your key";
        static string basePath = "https://api.e-iceblue.cn";
        static Configuration wordConfiguration = new Configuration(appId, appKey, basePath);
        static ShapesApi shapesApi = new ShapesApi(wordConfiguration);
        public static void AddShape()
        {         
            string name = "addShape.docx";
            string paragraphPath = "Section/0/Body/0/Paragraph/0";
            int indexInParagraph = 1;
            string folder = "input";
            string storage = null;
            string password = null;
            ShapeFormat shapeFormat = new ShapeFormat(40, 40, ShapeFormat.ShapeTypeEnum.Rectangle)
            {
                HorizontalOrigin = ShapeFormat.HorizontalOriginEnum.Page,
                VerticalOrigin = ShapeFormat.VerticalOriginEnum.Page,
                VerticalPosition = 200,
                HorizontalPosition = 100,
                FillColor = new Color(100, 100, 100),
                Rotation = 45,
                StrokeWeight = 2,
                StrokeColor = new Color(125, 125, 125),
                TextWrappingType = ShapeFormat.TextWrappingTypeEnum.Both,
                TextWrappingStyle = ShapeFormat.TextWrappingStyleEnum.InFrontOfText,
                ZOrder = 1
            };
            string destFilePath = "output/addShape_output.docx";
            shapesApi.AddShape(name, paragraphPath, shapeFormat, destFilePath,folder, storage, indexInParagraph, password);
        }

        public static void DeleteShape()
        {
            string name = "deleteShape.docx";
            string paragraphPath = "Section/0/Body/0/Paragraph/0";
            int? index = 0;
            string folder = "input";
            string storage = null;
            string password = null;
            string destFilePath = "output/deleteShape_output.docx";
            shapesApi.DeleteShape(name, paragraphPath, index, destFilePath, folder, storage, password);
        }
        public static void GetShapeFormat()
        {
            string name = "template.docx";
            string paragraphPath = "Section/0/Body/0/Paragraph/0"; ;
            int? index = 3;
            string folder = "input";
            string storage = null;
            string password = null;
            var response = shapesApi.GetShapeFormat(name, paragraphPath, index, folder, storage, password);
        }
        public static void GetShapes()
        {
            string name = "getShapes.docx";
            string folder = "input";
            string storage = null;
            string password = null;
            string paragraphPath = null;
            var response = shapesApi.GetShapes(name, folder, storage, password, paragraphPath);
        }
        public static void UpdateShapeFormat()
        {
            string name = "updateShapeFormat.docx";
            string paragraphPath = "Section/0/Body/0/Paragraph/0";
            int? index = 0;
            string folder = "input";
            string storage = null;
            string password = null;
            ShapeFormat shapeFormat = new ShapeFormat(40, 40, ShapeFormat.ShapeTypeEnum.Rectangle)
            {
                HorizontalOrigin = ShapeFormat.HorizontalOriginEnum.Page,
                VerticalOrigin = ShapeFormat.VerticalOriginEnum.Page,
                VerticalPosition = 200,
                HorizontalPosition = 100,
                FillColor = new Color(100, 100, 100),
                Rotation = 95,
                StrokeWeight = 1,
                StrokeColor = new Color(225, 15, 15),
                TextWrappingType = ShapeFormat.TextWrappingTypeEnum.Both,
                TextWrappingStyle = ShapeFormat.TextWrappingStyleEnum.InFrontOfText,
                ZOrder = 1
            };
            string destFilePath = "output/updateShapeFormat_output.docx";
            shapesApi.UpdateShapeFormat(name, paragraphPath, index, shapeFormat, destFilePath, folder, storage, password);
        }
    }
}
